packages/core/src/browser/quick-open/quick-command-service.ts

```diff
@@ -74,19 +90,102 @@ export class QuickCommandService implements QuickOpenModel, QuickOpenHandler {
         return { fuzzyMatchLabel: true };
     }
 
+    /**
+     * Get the list of recently used and other commands.
+     *
+     * @returns the list of recently used commands and other commands.
+     */
+    private getCommands(): { recent: Command[], other: Command[] } {
+
+        // Get the list of recent commands.
+        const recentCommands: Command[] = this.commands.recent;
+
+        // Get the list of all valid commands.
+        const allCommands: Command[] = this.getValidCommands(this.commands.commands);
+
+        // Build the list of recent commands.
+        const rCommands: Command[] = [];
+        recentCommands.forEach((r: Command) => {
+            const exists = [...allCommands].some((c: Command) => Command.equals(r, c));
+            // Add the recently used item to the list.
+            if (exists) { rCommands.push(r); }
+        });
+
+        // Build the list of other commands.
+        const oCommands: Command[] = [];
+        allCommands.forEach((a: Command) => {
+            const exists = [...rCommands].some((c: Command) => Command.equals(a, c));
+            // If the command does not exist in the recently used list, add it to the other list.
+            if (!exists) { oCommands.push(a); }
+        });
+
+        // Normalize the list of recent commands.
+        const recent: Command[] = this.normalize(rCommands);
+
+        // Normalize, and sort the list of other commands.
+        const other: Command[] = this.sort(
+            this.normalize(oCommands)
+        );
+
+        return { recent, other };
+    }
+
+    /**
+     * Normalizes a list of commands.
+     * Normalization includes obtaining commands that have labels and are visible.
+     *
+     * @param commands the list of commands.
+     * @returns the list of normalized commands.
+     */
+    private normalize(commands: Command[]): Command[] {
+        return commands
+            .filter((a: Command) => a.label)
+            .filter((a: Command) => this.commands.isVisible(a.id));
```

Comment:
could we do one `.filter()` instead ? i.e., `filter(a => !!a.label && isVisible(a.id))`